usaspending_api/search/delta_models/subaward_search.py

```diff
@@ -461,22 +465,40 @@
         fpds.product_or_service_code,
         psc.description AS product_or_service_description,
 
+        LATEST_CURRENT_CD.recipient_location_congressional_code_current AS legal_entity_congressional_current,
         COALESCE(UPPER(bs.sub_legal_entity_country_code), 'USA') AS sub_legal_entity_country_code,
         rcc.country_name AS sub_legal_entity_country_name,
+        -- For county codes, we need to make sure we extract the right digits out and ensure it's zero-padded to 3.
         LPAD(CAST(CAST(REGEXP_EXTRACT(rec.county_numeric, '^[A-Z]*(\\d+)(?:\\.\\d+)?$', 1) AS SHORT) AS STRING), 3, '0') AS sub_legal_entity_county_code,
         rec.county_name AS sub_legal_entity_county_name,
         LEFT(COALESCE(bs.sub_legal_entity_zip, ''), 5) AS sub_legal_entity_zip5,
         rec.census_code AS sub_legal_entity_city_code,
+        -- For congressional districts, we need to make sure we extract the right digits out and ensure it's zero-padded to 2.
         LPAD(CAST(CAST(REGEXP_EXTRACT(UPPER(bs.sub_legal_entity_congressional), '^[A-Z]*(\\d+)(?:\\.\\d+)?$', 1) AS SHORT) AS STRING), 2, '0') AS sub_legal_entity_congressional,
+        (CASE
+            WHEN (
+                UPPER(bs.sub_legal_entity_country_code) <> 'USA'
+            ) THEN NULL
+            ELSE COALESCE(rl_cd_state_grouped.congressional_district_no, rl_zips.congressional_district_no, rl_cd_zips_grouped.congressional_district_no, rl_cd_city_grouped.congressional_district_no, rl_cd_county_grouped.congressional_district_no)
+        END) AS sub_legal_entity_congressional_current,
 
+        LATEST_CURRENT_CD.pop_congressional_code_current AS place_of_performance_congressional_current,
         fabs.place_of_performance_scope AS place_of_perform_scope,
         COALESCE(UPPER(bs.sub_place_of_perform_country_co), 'USA') AS sub_place_of_perform_country_co,
         pcc.country_name AS sub_place_of_perform_country_name,
+        -- For county codes, we need to make sure we extract the right digits out and ensure it's zero-padded to 3.
         LPAD(CAST(CAST(REGEXP_EXTRACT(pop.county_numeric, '^[A-Z]*(\\d+)(?:\\.\\d+)?$', 1) AS SHORT) AS STRING), 3, '0') AS sub_place_of_perform_county_code,
         pop.county_name AS sub_place_of_perform_county_name,
         LEFT(COALESCE(sub_place_of_performance_zip, ''), 5) AS sub_place_of_perform_zip5,
         pop.census_code AS sub_place_of_perform_city_code,
-        LPAD(CAST(CAST(REGEXP_EXTRACT(UPPER(bs.sub_place_of_perform_congressio), '^[A-Z]*(\\d+)(?:\\.\\d+)?$', 1) AS SHORT) AS STRING), 2, '0') AS sub_place_of_perform_congressio
+        -- For congressional districts, we need to make sure we extract the right digits out and ensure it's zero-padded to 2.
```

Comment:
I think this is knowledge duplicatoin here. Looking at the code we can tell that this is happening

> we need to make sure we extract the right digits out and ensure it's zero-padded to 2.

but why?